render/gles2/texture.c

```diff
@@ -218,6 +218,18 @@ static void gles2_texture_get_matrix(struct wlr_texture *_texture,
 	wlr_matrix_mul(projection, matrix, matrix);
 }
 
+static void gles2_texture_get_buffer_size(struct wlr_texture *texture, struct
+		wl_resource *resource, int *width, int *height) {
+	struct wl_shm_buffer *buffer = wl_shm_buffer_get(resource);
+	if (!buffer) {
+		wlr_log(L_ERROR, "getting buffer size is only implemented for shm buffers");
```

Comment:
This isn't getting merged until it does DRM buffers too. Check out how the upload function handles it if you don't know how.